spanner/read.go

```diff
@@ -384,6 +384,7 @@ const (
 	queueingUnretryable                                    // 2
 	aborted                                                // 3
 	finished                                               // 4
+	ended                                                  // 5
```

Comment:
I don't understand why we would need a separate status for this, and can't just use finished.

Reply:
I added this to make sure we don't call cancel() while doing iter.Stop in case we saw the last message already